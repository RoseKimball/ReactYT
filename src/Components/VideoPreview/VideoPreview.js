import React, { Component } from 'react';
import './VideoPreview.scss';
import { Image } from 'semantic-ui-react';

export class VideoPreview extends Component {
    render() {
        const horizontal = this.props.horizontal ? 'horizontal' : null;
        return (
            <div className={['video-preview', horizontal].join(' ')}>
                <div className='image-container'>
                    <Image src='http://via.placeholder.com/210x118'/>
                    <div className='time-label'>
                        <span>07:45</span>
                    </div>
                </div>
                <div className='video-info'>
                    <div className='semi-bold two-lines-max'>Video Title</div>
                    <div className='metadata-container'>
                        <div className='channel-title'>Channel Title</div>
                        <div><span>2.1M views &middot; 2 days ago</span></div>
                    </div>
                </div>
            </div>
        )
    }

};

export default VideoPreview;