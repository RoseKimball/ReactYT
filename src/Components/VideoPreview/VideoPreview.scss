@import '../../styles/Shared.scss';

.video-preview {
    display: grid;
    grid: 118px auto / 210px;    //width of rows '/' width of columns
}

.image-container {
    position: relative;
    grid-row: 1 / 2;     // put first row between 1st and 2nd horizontal line
    grid-column: 1 / 2;  // put column btwn 1st and 2nd vertical line
}

.time-label {
    position: absolute;
    background-color: $text-color-dark;
    bottom: 0;
    right: 0;
    opacity: 0.8;
    border-radius: 2px;
    font-weight: 500;
    color: white;
    margin: 4px;
    padding: 2px 4px;
    line-height: 12px;
}

.video-info {
    color: $text-color-dark;

    .metadata-container {
        padding-top: 5px;
        font-size: 13px;
        color: #6e6e6e;

        .channel-title {
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
        }
    }

    .two-lines-max {
        overflow: hidden;
        line-height: 1.4em;
        max-height: 2.8em;
    }

    .semi-bold {
        font-weight: 600;
    }
}

