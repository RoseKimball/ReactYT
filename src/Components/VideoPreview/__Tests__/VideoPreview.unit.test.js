import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import VideoPreview from '../VideoPreview';

describe('VideoPreview', () => {
    test('renders VideoPreview', () => {
        const wrapper = shallow (
            <VideoPreview />
        )
        expect(wrapper).toMatchSnapshot();
    })
    test('renders horizontal preview', () => {
        const wrapper = shallow (
            <VideoPreview horizontal={true}/>
        )
        expect(wrapper).toMatchSnapshot();
    })
})