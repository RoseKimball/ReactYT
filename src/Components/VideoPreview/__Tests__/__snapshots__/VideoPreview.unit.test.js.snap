// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPreview renders VideoPreview 1`] = `ShallowWrapper {}`;

exports[`VideoPreview renders horizontal preview 1`] = `ShallowWrapper {}`;
